### PR TITLE
completed-react-effects-Bishan-Webeshete

### PR DESCRIPTION
**Pull request by BishanWebeshete**
Kapture 2023-04-02 at 20 00 38

### CODE REVIEW

File: react-effects/my-app/src/List.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  useEffect(()=> {
+    if (isLoading) {
+      const promise=readItems();
+      promise.then((value)=>{
+        setItems(value);
+        setIsLoading(false);
+      });
+      promise.catch((value)=> {
+        setError(value);
```
Comment (RobertGardner) [inline react-effects/my-app/src/List.js:23]: set isLoading to false here so the Loading... message doesn't stay there when there's an error.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: react-effects/my-app/src/List.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      });
+    }
+  }, [isLoading])
```
[the remark this answers is not in the document]
Comment (BishanWebeshete) [reply react-effects/my-app/src/List.js:26]: I got a linter warning when trying to leave the array empty so I kept isLoading in my dependencies.

**Comment by BishanWebeshete** [comment]
updated react-effects to set isLoading properly

File: react-effects/my-app/src/List.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (RobertGardner) [reply react-effects/my-app/src/List.js:27]: Oh, that's correct because you check `if (isLoading)` in your Effect. Well done.

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**